src/ui/page/share/dataset/steps/dataSelection/dataSelection.tsx

```diff
@@ -7,8 +9,36 @@ import { FilePicker } from '@/ui/component/filePicker/filePicker'
 import { type Item, List } from '@/ui/component/list/list'
 import { Icon } from '@/ui/component/icon/icon'
 import { useFileStore } from '@/ui/store'
-import type { StoreFilesInput } from '@/domain/file/command'
+import type {
+  FileId,
+  ResourceConflictError,
+  ResourceNotFoundError,
+  StoreFilesInput
+} from '@/domain/file/command'
 import './dataSelection.scss'
+import {
+  useDispatchNotification,
+  type DispatchNotificationInput
+} from '@/ui/hook/useDispatchNotification'
+
+type FileError = ResourceConflictError | ResourceNotFoundError
+
+const logFileError = (error: FileError): void => {
+  switch (error._tag) {
+    case 'resource-conflict': {
+      console.error(
+        "Oops... You are trying either to store a file whose id already exists in memory or to store files with the same id... So we can't store these files..."
```

Comment:
The most canonical way would be to use the [Show](https://gcanti.github.io/fp-ts/modules/Show.ts.html#show-overview) class type, and have it implemented by errors. Then, to have an IO for logging (but you can always use an effectful function if you prefer)